rest/data/distro.go

```diff
@@ -69,13 +70,19 @@ func (dc *DBDistroConnector) CreateDistro(distro *distro.Distro) error {
 
 // DeleteDistroById removes a given distro from the database based on its id.
 func (dc *DBDistroConnector) DeleteDistroById(distroId string) error {
-	err := distro.Remove(distroId)
-	if err != nil {
+	var err error
+	if err = distro.Remove(distroId); err != nil {
 		return gimlet.ErrorResponse{
 			StatusCode: http.StatusInternalServerError,
 			Message:    fmt.Sprintf("distro with id '%s' was not deleted", distroId),
 		}
 	}
+	if err = host.MarkInactiveStaticHosts([]string{}, distroId); err != nil {
```

Comment:
If the host cleanup fails do we want the distro to be removed? 
In other words, do we want to remove the hosts and then remove the distro so we're sure there are no orphaned hosts?

Reply:
I inverted the cases.